app/controllers/routes_controller.rb

```diff
@@ -11,9 +11,13 @@ def show
 
   def update
     route_details = @request_data[:route]
-    @route = Route.find_or_initialize_by(incoming_path: route_details.delete(:incoming_path))
-    status_code = @route.new_record? ? 201 : 200
-    @route.update_attributes(route_details) || status_code = 422
+    @route = Route.new(route_details)
+    if @route.valid?
```

Comment:
Well we could just put `unless: :new_record?` there which would make it even simpler, but as you say I'm not sure what that would give us.